main.go

```diff
@@ -70,6 +71,7 @@ func main() {
 
 	opts := zap.Options{
 		TimeEncoder: zapcore.RFC3339NanoTimeEncoder,
+		ZapOpts:     []zaplog.Option{zaplog.AddCaller(), zaplog.AddCallerSkip(-1)},
```

Comment:
What's the skip for?

Reply:
we cannot get the right `caller` if we do not add `zaplog.AddCallerSkip(-1)`.
we get some logs like this:
`"caller":"controller/controller.go:114","msg":"Job is suspended and workload not yet admitted by a clusterQueue, nothing to do",`
In fact this log should be printed at `pkg/controller/workload/job/job_controller.go:208`.

Reply:
What about other log lines? For example, the ones coming from the scheduler.

I'm wondering if controller-runtime is increasing the Skip somewhere just when calling Reconcile.

Reply:
the controller-runtime will add 1 Skip in [controller-runtime/pkg/log/zap/zap.go](https://github.com/kubernetes-sigs/controller-runtime/blob/26c95adfec6390a55f76b2c3ca73ffb532b12dec/pkg/log/zap/zap.go#L247). Maybe we should descrease 1?
there is an issue to track this problem: https://github.com/kubernetes-sigs/controller-runtime/issues/1737

Reply:
logs in scheduler.go: 
`{"level":"Level(-2)","ts":"2022-05-10T02:37:06.627754573Z","logger":"scheduler","caller":"scheduler/scheduler.go:306","msg":"Workload assumed in the cache","workload":{"name":"sample-job-hw88n","namespace":"default"},"clusterQueue":{"name":"cluster-total"}}
{"level":"Level(-3)","ts":"2022-05-10T02:37:06.627876656Z","logger":"scheduler","caller":"scheduler/scheduler.go:137","msg":"Workload evaluated for admission","workload":{"name":"sample-job-hw88n","namespace":"default"},"clusterQueue":{"name":"cluster-total"},"status":"assumed","reason":""}`

Reply:
Are those logs with the `-1` you added?

If so, then I'm fine leaving it and then we can remove it once https://github.com/kubernetes-sigs/controller-runtime/issues/1737 is fixed.

Reply:
> Are those logs with the `-1` you added?

These logs are not added by me. Line number was not correct because I pushed these code based on the old commit(e5fdd9b3a14259ad89e19e7cd2b71dabc96eac30)

Reply:
Let me restate the question:

Do the logs for scheduler.go file look accurate when using the code in this PR?

Reply:
yes. 
https://github.com/kubernetes-sigs/kueue/blob/e5fdd9b3a14259ad89e19e7cd2b71dabc96eac30/pkg/scheduler/scheduler.go#L303-L307